### ISSUE TITLE
[websockets] Upgrade header value case sensitive

### ISSUE THREAD

**Issue by mathiasertl** [processone/ejabberd#510]
When testing websocket support, I found that the `Upgrade` header value is treated as case sensitive. For example:

```
$ curl -i -N -H "Host: webchat.jabber.at" -H "Origin: http://webchat.jabber.at" \
    -H "Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits" \
    -H "Sec-WebSocket-Key: abc" -H "Sec-WebSocket-Protocol: xmpp" \
    -H "Sec-WebSocket-Version:13" \
    -H "Connection: Upgrade" -H "Upgrade: websocket"  http://localhost:5280/ws/
HTTP/1.1 101 Switching Protocols
...
```

While the following doesn't work. Note the "Upgrade: **W**eb**S**ocket" casing in the last header:

```
$ curl -i -N -H "Host: webchat.jabber.at" -H "Origin: http://webchat.jabber.at" \
    -H "Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits" \
    -H "Sec-WebSocket-Key: abc" -H "Sec-WebSocket-Protocol: xmpp" \
    -H "Sec-WebSocket-Version:13" \
    -H "Connection: Upgrade" -H "Upgrade: WebSocket"  http://localhost:5280/ws/
HTTP/1.1 200 OK
...
...This web page is only informative. ...
```

The problem is that at least when using Apaches [mod_proxy_wstunnel](http://httpd.apache.org/docs/2.4/mod/mod_proxy_wstunnel.html), requests are send by Apache using the latter. I am not really sure if Apaches behavior conforms to standards: [RFC6455](http://tools.ietf.org/html/rfc6455) always uses `websocket` as value, but [Section 11.2](http://tools.ietf.org/html/rfc6455#section-11.2) registers `WebSocket` as Upgrade keyword.

Could ejabberd ignore case when parsing the ```Upgrade``` header?

**Comment by weiss** [comment]
> I am not really sure if Apaches behavior conforms to standards

I think it does, as [RFC 6455, section 4.2.1][1] clearly says that this header value must be "treated as an ASCII case-insensitive value".  I've submitted a fix in PR #513.

[1]: https://tools.ietf.org/html/rfc6455#section-4.2.1

**Comment by mathiasertl** [comment]
We have this patch deployed at jabber.at now and it does seem to improve the situation. It does connect now albeit still with some problems, which I'll file a separate issue for.
